WordPress/src/main/java/org/wordpress/android/networking/menus/MenusRestWPCom.java

```diff
@@ -0,0 +1,162 @@
+    private static final String DELETE_MENU_REST_PATH = "/sites/%s/menus/%s/delete";
+
+    //
+    // JSON keys for fetchAllMenus response object
+    //
+    public static final String MENU_KEY = "menu";
+    public static final String ALL_MENUS_MENUS_KEY = "menus";
+
+    private static final String DELETED_KEY = "deleted";
+
+    private final IMenusDelegate mDelegate;
+
+    public MenusRestWPCom(IMenusDelegate delegate) {
+        if (delegate == null) {
+            throw new IllegalArgumentException("IMenusDelegate cannot be null");
+        }
+        mDelegate = delegate;
+    }
+
+    public boolean createMenu(final MenuModel menu) {
+        // a name string is required for the REST call
+        if (menu == null || TextUtils.isEmpty(menu.name)) return false;
+        String path = formatPath(CREATE_MENU_REST_PATH, null);
+        Map<String, String> params = new HashMap<>();
+        params.put(MENU_NAME_KEY, menu.name);
+        WordPress.getRestClientUtilsV1_1().post(path, params, null, new RestRequest.Listener() {
+            @Override public void onResponse(JSONObject response) {
+                menu.menuId = response.optLong(MENU_ID_KEY);
+                mDelegate.onMenuCreated(HttpURLConnection.HTTP_OK, menu);
+                // TODO: call updateMenu if the menu has any non-default fields
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override public void onErrorResponse(VolleyError error) {
+                mDelegate.onMenuCreated(statusCodeFromVolleyError(error), menu);
+            }
+        });
+        return true;
+    }
+
+    public boolean updateMenu(final MenuModel menu) {
+        if (menu == null) return false;
+        String path = formatPath(MENU_REST_PATH, String.valueOf(menu.menuId));
+        Map<String, String> params = new HashMap<>();
+        params.put(MENU_NAME_KEY, menu.name);
+        WordPress.getRestClientUtilsV1_1().post(path, params, null, new RestRequest.Listener() {
+            @Override public void onResponse(JSONObject response) {
+                MenuModel menu = menuFromJson(response.optJSONObject(MENU_KEY));
+                mDelegate.onMenuUpdated(HttpURLConnection.HTTP_OK, menu);
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override public void onErrorResponse(VolleyError error) {
+                mDelegate.onMenuUpdated(statusCodeFromVolleyError(error), menu);
+            }
+        });
+        return true;
+    }
+
+    public void fetchMenu(long menuId) {
+        String path = formatPath(MENU_REST_PATH, String.valueOf(menuId));
+        Map<String, String> params = new HashMap<>();
+        WordPress.getRestClientUtilsV1_1().get(path, params, null, new RestRequest.Listener() {
+            @Override public void onResponse(JSONObject response) {
+                JSONObject menuObject = response.optJSONObject(MENU_KEY);
+                mDelegate.onMenuReceived(HttpURLConnection.HTTP_OK, menuFromJson(menuObject));
```

Comment:
Awesome! yes I follow what you say in the sense I can imagine what the intention of the server API is, but as it is right now it doesn't follow the REST principles strictly. I'm not a purist myself, but just worth observing this as it might leave some decisions hanging in the air if not agreed upon first.